spacy/strings.pyx

```diff
@@ -291,27 +193,141 @@ cdef class StringStore:
             self.add(word)
         return self
 
+    def _transform_to_list(self, function: Callable[[str, int], Any]) -> Collection[Any]:
```

Comment:
This seems a bit overengineered with the `Callable` for my taste? I think it's okay to just implement the individual `keys` and `values` methods without quite as much generalization?